lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java

```diff
@@ -66,38 +67,52 @@ public void setFragmentManager(FragmentManager fragmentManager) {
     this.fragmentManager = fragmentManager;
   }
 
-  @Override
-  public void addView(View child, int index, ViewGroup.LayoutParams params) {
-    int backStackEntryCount = fragmentManager.getBackStackEntryCount();
-    reverseDrawing = backStackEntryCount > previousBackStackEntryCount;
-    previousBackStackEntryCount = backStackEntryCount;
-    super.addView(child, index, params);
+  public void willDetachCurrentScreen() {
+    isDetachingCurrentScreen = true;
   }
 
   @Override
-  public void removeView(View view) {
+  public void removeView(final View view) {
+    if (isDetachingCurrentScreen) {
+      isDetachingCurrentScreen = false;
+      reverseLastTwoChildren = true;
+    }
+
     super.removeView(view);
   }
 
+  private void drawAndRelease(int index) {
+    DrawingOp op = drawingOps.remove(index);
+    op.draw();
+    drawingOpPool.add(op);
+  }
+
   @Override
   protected void dispatchDraw(Canvas canvas) {
     super.dispatchDraw(canvas);
 
+	// check the view removal is completed (by comparing the previous children count)
+    if (drawingOps.size() < previousChildrenCount) {
+      reverseLastTwoChildren = false;
+    }
+    previousChildrenCount = drawingOps.size();
+
+    if (reverseLastTwoChildren) {
```

Comment:
@Igor1201 Would this be even more readable if you did an explicit swap of the last 2 items in the if block then just loop through them all?